sources/packages/web/src/components/common/restriction/ViewRestriction.vue

```diff
@@ -73,19 +73,14 @@
         </template>
       </template>
       <template #footer>
-        <check-permission-role :role="allowedRole">
-          <template #="{ notAllowed }">
-            <footer-buttons
-              :processing="processing"
-              :primaryLabel="allowUserToEdit ? 'Resolve restriction' : 'Close'"
-              secondaryLabel="Cancel"
-              @primaryClick="allowUserToEdit ? submit() : cancel()"
-              @secondaryClick="cancel"
-              :disablePrimaryButton="notAllowed"
-              :showSecondaryButton="allowUserToEdit"
-            />
-          </template>
-        </check-permission-role>
+        <footer-buttons
+          :processing="processing"
+          :primaryLabel="allowUserToEdit ? 'Resolve restriction' : 'Close'"
+          secondaryLabel="Cancel"
+          @primaryClick="allowUserToEdit ? submit() : cancel()"
+          @secondaryClick="cancel"
+          :showSecondaryButton="allowUserToEdit"
```

Comment:
Or as a bare minimal solution. 

![image](https://github.com/user-attachments/assets/53b58676-a16e-40dc-8682-1a12e509a96f)

Reply:
I agree with this solution, which is adopted in the latest commit.